gnat2goto/driver/tree_walk.adb

```diff
@@ -880,6 +894,97 @@ package body Tree_Walk is
       return Do_Bare_Range_Constraint (N, Underlying);
    end Do_Array_Range;
 
+   ------------------
+   -- Add_To_Table --
+   ------------------
+
+   function Add_To_Table (Name : String; Symbol_Type : Irep; Value : Irep)
+                           return Symbol is
+      New_Symbol : Symbol;
+   begin
+      New_Symbol.SymType := Symbol_Type;
+      New_Symbol.Name := Intern (Name);
+      New_Symbol.PrettyName := New_Symbol.Name;
+      New_Symbol.BaseName := New_Symbol.Name;
+      New_Symbol.Mode := Intern ("C");
+      New_Symbol.Value := Value;
+
+      if not (Global_Symbol_Table.Contains (Key => Intern (Name))) then
+         Global_Symbol_Table.Insert (Intern (Name), New_Symbol);
+      end if;
+      return New_Symbol;
+   end Add_To_Table;
+
+   function Make_Range_Assert_Expr (N : Node_Id; Value : Irep;
+                                                 Value_Type : Irep)
+                                    return Irep
+   is
+      Call_Inst : constant Irep := New_Irep (I_Side_Effect_Expr_Function_Call);
+      Call_Args : constant Irep := New_Irep (I_Argument_List);
+      Actual_Type : constant Irep := Follow_Symbol_Type (Get_Type (Value),
+                                                         Global_Symbol_Table);
+
+      function Build_Assert_Function return Symbol;
+
+      ---------------------------
+      -- Build_Assert_Function --
+      ---------------------------
+
+      function Build_Assert_Function return Symbol
+      is
+         Func_Type : constant Irep := New_Irep (I_Code_Type);
+         Func_Args : constant Irep := New_Irep (I_Parameter_List);
+         Func_Name : constant String :=
+           Fresh_Var_Name ("range_check");
+         Body_Block : constant Irep := New_Irep (I_Code_Block);
```

Comment:
:pick: I'd prefer `Make_XXX` functions in place of these

Reply:
Done.